src/reanimated2/component/FlatList.tsx

```diff
@@ -79,8 +69,8 @@ export const ReanimatedFlatList = forwardRef(
     }
 
     const cellRenderer = React.useMemo(
-      () => createCellRenderer(itemLayoutAnimation, cellStyle),
-      [cellStyle]
+      () => createCellRenderer(itemLayoutAnimation),
+      []
```

Comment:
why is it empty

Reply:
Adding `itemLayoutAnimation` to the dependency array would cause issues e.g. if we had `itemLayoutAnimation={Layout.springify()}`, every time the FlatList rerenders (e.g. when an item was added) a new `Layout` object would be created causing a new `CellRendererComponent` to be created. Since every element of the list is wrapped with `CellRendererComponent` it would make all items exit and reenter on every render.

So normally for 
```js
<Animated.FlatList
          itemLayoutAnimation={Layout.springify()}
          ...
/>
```
we have:

https://github.com/software-mansion/react-native-reanimated/assets/56109050/8ceb1b5e-f77e-4c5f-beee-ce7d7a4b3a1d

but if we changed our implementation to
```js
      () => createCellRenderer(itemLayoutAnimation),
      [itemLayoutAnimation]

```
we would have:

https://github.com/software-mansion/react-native-reanimated/assets/56109050/fa1da14f-5426-4ae8-89c6-fab07a030ddf

Reply:
okay, makes sense to me 👍 tanks for clarification: <img src="https://github.com/software-mansion/react-native-reanimated/assets/20516055/6fdb053b-be70-4a65-8d54-a12521d734cb" height="20" /><img src="https://github.com/software-mansion/react-native-reanimated/assets/20516055/6fdb053b-be70-4a65-8d54-a12521d734cb" height="20" /><img src="https://github.com/software-mansion/react-native-reanimated/assets/20516055/6fdb053b-be70-4a65-8d54-a12521d734cb" height="20" />